### ISSUE TITLE
Outputs are always placed in the home root

### ISSUE THREAD

**Issue by arkjinli** [thomasjo/atom-latex#87]
The Atom version is 0.201.0, and I am using Ubuntu. The problem is: outputs are always placed in the home root, rather than where the .tex file is

This is the first time I had this problem, just after I updated the editor and the latex package.

The console error is "TeXification failed with status code 12".

Can you help me? Thank you very much!

**Comment by thomasjo** [comment]
If you could provide me with some additional information, that'd be brilliant;
  1. The output of executing `apm list --installed` from a terminal.
  2. Have you specified a value for the **Output Directory** (`latex.outputDirectory`) option?
  3. The contents of a `.log` file associated with your issue.

Thanks in advance.

**Comment by arkjinli** [comment]
Hi, sorry for the late reply.

Here is the information you want:

1. The out put of executing `apm list --installed` from a terminal:
    ```
    /home/ark/.atom/packages (4)
    ├── language-latex@0.6.1
    ├── latex@0.24.0
    ├── pdf-view@0.22.0
    └── script@2.23.0
    ```
    I am a new user to Atom so I didn't install many packages.

2. Have I specified a value for the **Output Directory** option?
    **No**, I leave the output directory blank.

3. The contents of a `.log` file. I created a minimum example which is basically a `hello world!` file:
    ```latex
    \documentclass{article}

    \begin{document}
        Hello world!
    \end{document}
    ```
    The corresponding `.log` file is
    ```
    This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2015/dev/Debian) (preloaded format=pdflatex 2015.5.29)  29 MAY 2015 23:05
    entering extended mode
     restricted \write18 enabled.
     file:line:error style messages enabled.
     %&-line parsing enabled.
    **helloworld.tex
    (./helloworld.tex
    LaTeX2e <2014/05/01>
    Babel <3.9l> and hyphenation patterns for 79 languages loaded.
    (/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
    Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
    (/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo
    File: size10.clo 2014/09/29 v1.4h Standard LaTeX file (size option)
    )
    \c@part=\count79
    \c@section=\count80
    \c@subsection=\count81
    \c@subsubsection=\count82
    \c@paragraph=\count83
    \c@subparagraph=\count84
    \c@figure=\count85
    \c@table=\count86
    \abovecaptionskip=\skip41
    \belowcaptionskip=\skip42
    \bibindent=\dimen102
    ) (/home/ark/helloworld.aux)
    \openout1 = `helloworld.aux'.

    LaTeX Font Info:    Checking defaults for OML/cmm/m/it on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
    LaTeX Font Info:    Checking defaults for T1/cmr/m/n on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
    LaTeX Font Info:    Checking defaults for OT1/cmr/m/n on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
    LaTeX Font Info:    Checking defaults for OMS/cmsy/m/n on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
    LaTeX Font Info:    Checking defaults for OMX/cmex/m/n on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
    LaTeX Font Info:    Checking defaults for U/cmr/m/n on input line 3.
    LaTeX Font Info:    ... okay on input line 3.
     [1

    {/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (/home/ark/helloworld.aux) ) 
    Here is how much of TeX's memory you used:
     202 strings out of 493105
     2255 string characters out of 6137072
     52508 words of memory out of 5000000
     3752 multiletter control sequences out of 15000+600000
     3640 words of font info for 14 fonts, out of 8000000 for 9000
     1141 hyphenation exceptions out of 8191
     23i,4n,17p,118b,107s stack positions out of 5000i,500n,10000p,200000b,80000s
    </usr/share/texlive/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb>
    Output written on /home/ark/helloworld.pdf (1 page, 11935 bytes).
    PDF statistics:
     12 PDF objects out of 1000 (max. 8388607)
     7 compressed objects within 1 object stream
     0 named destinations out of 1000 (max. 500000)
     1 words of extra memory for PDF output out of 10000 (max. 10000000)
    ```
    I tried this and it still outputs everything to my home directory :(.

FYI, the Ubuntu system I am using is 15.04 64-bit.

Thank you!

**Comment by thomasjo** [comment]
Thanks for additional info! I'm not really sure what's going wrong, but it's almost as if the current working directory incorrectly gets set to your `$HOME` directory. Unless I'm able to reproduce, I might have to get you to test out custom debug branch. Will let you know. Sorry about the trouble!

**Comment by arkjinli** [comment]
Thanks for your suggestive comment!

You are right. When I start atom at a different place, for example, `~/Document$ atom`, the output file goes to `~/Document`. It seems like the current working directory does not change properly to the place where the `.tex` file is, but is always fixed to where the atom session is called (`$pwd`).

I check the `\usr\bin\atom`. For Linux it starts by running `nohup atom --executed-from "$(pwd)"` which does depend on `$pwd`. If I start it from the Unity launcher it is `$HOME`.

I guess the default output directory is somehow related to where the atom starts up, instead of where your current working directory is in the project tree view. That makes sense to me.

I hope this new information will be helpful. It's really strange that only I get this bug. Is it possibly because of the new version of Ubuntu... I don't know.

Many thanks!

**Comment by Victorystick** [comment]
You're not alone. I have the same issue. I originally thought the problem was related issue #71, since I couldn't find the generated PDF file. I'm also running Atom on Ubuntu 15.04 64-bit.

**Comment by thomasjo** [comment]
Sorry about the trouble! I'm setting up an Ubuntu 15.04 VM now to attempt to reproduce the issue. Until I'm able to reproduce the problem, I'm just fumbling around in the blind.

**Comment by thomasjo** [comment]
I just managed to reproduce the issue. Will try to find time to resolve it tonight.

**Comment by Lednerb** [comment]
Same problem here. Again on a fresh-installed Ubuntu 15.04 64-Bit
Running latest Atom 0.209.0

Are there any workarounds or fixes so far?

**Comment by arkjinli** [comment]
One workaround is to start Atom from command line at the directory where your latex file is (or where you want the output file to be). It seems the problem is the latex package will always treat the place where Atom is started as the output directory.

**Comment by thomasjo** [comment]
I'm working on a fix, and I originally wanted to find a clean solution, but I might just have to implement platform specific hacks for the time being. Then later try to resolve the issue in a cleaner manner.

**Comment by Lednerb** [comment]
Seems to be fixed.
Only one little thing, but maybe it's caused by another plugin "project-manager".

Compiling the .tex file within a subdir, the compiled data will be saved in the project root dir.

Here's an example:

![folderstructure](https://cloud.githubusercontent.com/assets/2056876/8233915/32548622-15d9-11e5-95d4-ab91d912794f.png)

I've compiled the 'Text.tex' file.

**Comment by Lednerb** [comment]
There's another problem:
The compilation succeed within the parent directory, but in the status bar an error is displayed. Also a re-compilation does not work.

![error](https://cloud.githubusercontent.com/assets/2056876/8233968/8b204282-15d9-11e5-8554-a068da6b323a.png)

**Comment by thomasjo** [comment]
@Lednerb Thanks for the info! The first issue is actually how I implemented this, but I guess it makes more sense for the "root folder" to be the directory containing the (root) `.tex` file instead? That is easily fixed, and actually simplifies the logic in [`Composer::resolveProjectPath()`](https://github.com/thomasjo/atom-latex/blob/tj-fix-cwd-issue/lib/composer.js#L139-L154).

It's almost tempting to add a configuration option for this, whether to use the containing directory or the project directory as the target for build output. Does that make sense?

**Comment by Lednerb** [comment]
For my own projects i handle it normally the way you described it: in the project root is also the root '.tex', so the structure is more clearly but my girlfriend does it this way.

I think an option for this would be very nice. (default is root folder :wink:)

**Comment by thomasjo** [comment]
Just published a patch release (v0.24.2) that should resolve this once and for all.

**Comment by Poeschl** [comment]
I'm sry to jump in here again, but it seems that the texification with error number 12 is back.

My system:
OS: Linux Mint 17.1
TeX: TeX Live 2015
Atom: 1.2.4

So I changed my build path for latex to a different folder and checked the option to move the generated pdf to the tex document.
It worked fine this way, but since a few days every second build the plugin shows me an compilation error. So I investigated a little and in the console I get the "TeXification failed with status code 12" error ,it also seems that the pdf gets generated correctly but will not be moved to the location of the .tex file.

So for now compiling it a second time works out, but it is only a workaround.
